Add smart library demo

ivanvikhrev opened the pull request.

Commit:
Add colorization

Commit:
demos: fix cppcheck warnings about reducible variable scope

Commit:
demos: fix performance warnings from Cppcheck

Commit:
demos/object_detection_demo_faster_rcnn: untie DetectionOutputPostProcessor from the network

The code makes it seem like the postprocessor is somehow integrated with
the network, but it really isn't. It's called separately to convert
the output into a more palatable representation.

Removing all of the pseudo-integration makes the code a lot simpler.

Commit:
demos/object_detection_demo_faster_rcnn: turn most parameters into constants

And remove the boolean/enum parameters entirely. They just create unreachable
code.

ivanvikhrev commented:
@valentina-kustikova

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+
+### How it works
+
+The application is started from command line. It accepted several parameters. 
```
It accepts

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+
+The application is started from command line. It accepted several parameters. 
+It reads video stream frame-by-frame from a web-camera device and performs independent analysis
+of each frame. To make predictions the application using 3 models. An input frame is processed by
```
the application uses

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+The application is started from command line. It accepted several parameters. 
+It reads video stream frame-by-frame from a web-camera device and performs independent analysis
+of each frame. To make predictions the application using 3 models. An input frame is processed by
+the face detection model to predict face bounding boxes. Then, face keypoints
```
to detect bounding boxes.

valentina-kustikova replied:
После Then не нужна запятая

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+of each frame. To make predictions the application using 3 models. An input frame is processed by
+the face detection model to predict face bounding boxes. Then, face keypoints
+are predicted by the facial landmarks regression model. Keypoints are using
+to align the face and match it with face in data base. 
```
database

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+are predicted by the facial landmarks regression model. Keypoints are using
+to align the face and match it with face in data base. 
+To register in library press `r` on keyboard. Once reader has beeen registered, he will be autothorized 
+through face recognition. Registration allows to receiving and ruturning books by recognizing QR codes for 
```
ruturning -> returning

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+to align the face and match it with face in data base. 
+To register in library press `r` on keyboard. Once reader has beeen registered, he will be autothorized 
+through face recognition. Registration allows to receiving and ruturning books by recognizing QR codes for 
+each book. To make book recognition press `b` on keyboard. Also applications provides some 
```
Also application без s на конце

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+through face recognition. Registration allows to receiving and ruturning books by recognizing QR codes for 
+each book. To make book recognition press `b` on keyboard. Also applications provides some 
+extra statictics in console, like list of registered readers, full list of books in the library,
+hystory of borrowing books. To change information in console press `f` on keyboard. To exit press `q`.
```
hystory -> history

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+extra statictics in console, like list of registered readers, full list of books in the library,
+hystory of borrowing books. To change information in console press `f` on keyboard. To exit press `q`.
+
+### Creating QR-codes for books
```
вставить пустую строку после заголовка

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+hystory of borrowing books. To change information in console press `f` on keyboard. To exit press `q`.
+
+### Creating QR-codes for books
+Next two files are using to create QR-codes for books:
```
The following files are used ...

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+### Creating QR-codes for books
+Next two files are using to create QR-codes for books:
+`library.json` file contains information about books in the library. 
+`createQRCodes.py` script generates QR-codes for each book in `library.json` file.
```
сделать маркированный список

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+optional arguments:
+  -h, --help  show this help message and exit
+  -i LIB      unput `.json` file with info
+  -o OUT      directory to save generated QR-codes
```
Вот этот блок или его части не должны быть оформлены как код?

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
```diff
@@ -0,0 +1,131 @@
+pip install -r requirements.txt
+
+
+### Running the demo:
```
Удалить двоеточие в конце заголовка

valentina-kustikova commented on demos/python_demos/smart_library_demo/README.md:
````diff
@@ -0,0 +1,131 @@
+
+Linux (`sh`, `bash`, ...) (assuming OpenVINO installed in `/opt/intel/openvino`):
+
+``` sh
````
не нашла конца предыдущего блока, поэтому отображается неправильно

valentina-kustikova commented on demos/python_demos/smart_library_demo/createQRCodes.py:
```diff
@@ -0,0 +1,39 @@
+    parser.add_argument('-i', type=str, dest = 'lib',  default='library.json')
+    parser.add_argument('-o', type=str, dest = 'out',  default='qr-codes')
+    return parser.parse_args()
+def main():
```
А это нормально, что функции не отделены пустой строкой, а потом в 16-17 две пустых строки? Надо проверить стиль кодирования.

valentina-kustikova commented on demos/python_demos/smart_library_demo/createQRCodes.py:
```diff
@@ -0,0 +1,39 @@
+    gen = qr.QRgenerator()
+
+
+    if (args.lib != None and  os.path.isfile(args.lib) and args.out != None ):
```
Удалить пробел после None в конце строки.

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+def createArgparse():
+    parser = argparse.ArgumentParser(description='Smart Library Sample')
+    parser.add_argument('-reid', type = str,#default = 'DNNfr',
+                        dest = 'rdDet', required=True, help = 'Required. Type of recognizer. Available DNN face recognizer - DNNfr')
```
Аналогичный вопрос по стилю кодирования.
Больше 80 символов в строку обычно не делают.
Потом где-то знак равенства окружен пробелами, а где-то - нет.
Наличие закомментированных значений по умолчанию тоже выглядит странно.

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+    parser.add_argument('-m_lm', type = str, #default = 'landmarks-regression-retail-0009.xml',
+                        dest = 'lmModel', required=True, help = 'Required. Path to .xml file')    
+
+    parser.add_argument('-w_rd', type = int, default = '128',
```
Странно, что значение по умолчанию в одинарных кавычках, хотя вроде бы целочисленный тип. Не уверена, что текущий вариант правильный.

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+    textSize = cv.getTextSize(text, font, scale, thickness) 
+    if rect:
+        cv.rectangle(img, pos, (textSize[0][0] + ix, 
+                pos[1]-textSize[0][1] + iy), (255, 255, 255), cv.FILLED)
```
Вокруг минуса нужны пробелы

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+    if rect:
+        cv.rectangle(img, pos, (textSize[0][0] + ix, 
+                pos[1]-textSize[0][1] + iy), (255, 255, 255), cv.FILLED)
+    cv.putText(img, text, (pos[0], pos[1]+iy),
```
Вокруг плюса нужны пробелы

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+        cv.rectangle(img, pos, (textSize[0][0] + ix, 
+                pos[1]-textSize[0][1] + iy), (255, 255, 255), cv.FILLED)
+    cv.putText(img, text, (pos[0], pos[1]+iy),
+                font, scale, color, thickness)
```
форматирование поехало

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+    indent += txtSize[0][1] + 5
+    text = 'r - register'
+    putText(img, text, (5,  indent), 0, 0, cv.FONT_HERSHEY_PLAIN, 
+                        (22, 163, 245), 1, 1, 0)
```
поехало форматирование

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+            userID = int(np.argmax(out) + 1)
+            text = 'User #' + str(userID)
+            putText(img, text, face[0], face[0][0], -5, cv.FONT_HERSHEY_SIMPLEX, 
+                                           (22, 163, 245), 1, 2)
```
поехало форматирование

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+        else:
+            text = 'Unknown'
+            putText(img, text, face[0], face[0][0], -5, cv.FONT_HERSHEY_SIMPLEX, 
+                                           (22, 163, 245), 1, 2)     
```
поехало форматирование

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+        return bID 
+    except ValueError:
+        return -1
+def main():
```
вставить пустую строку перед функцией

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+    data = bookRec.recognize(img)
+    try: 
+        bID = int(data.split(' ')[0])
+        print(bID)
```
У меня вопрос, каким образом делается логирование в стандартных семплах. Надо использовать для логирования те же инструменты.

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+    putText(img, text, (5,  indent), 0, 0, cv.FONT_HERSHEY_PLAIN, 
+            (22, 163, 245), 1, 1, 0)
+
+def recUser(img):
```
Надо проверить, что написание названий функций соответствует принятому в библиотеке стилю кодирования.

valentina-kustikova commented on demos/python_demos/smart_library_demo/smart_library_demo.py:
```diff
@@ -0,0 +1,233 @@
+    DB = DynamicDB()
+
+    if (args.rdDet != None and args.fdDet != None and args.lmDet != None):
+        rdArgs ['name'] = args.rdDet
```
Не уверена, что стиль кодирования семплов предусматривает пробел перед квадратной скобкой. Надо в целом проверить стиль кодирования, а то у нас зоопарк по семплу.

valentina-kustikova commented on demos/python_demos/smart_library_demo/src/modules/face_recognizer.py:
```diff
@@ -0,0 +1,211 @@
+        return self.counter
+    
+    
+
```
Надо удалить лишние пустые строки

Commit:
Correct data

Commit:
Correct data 2

Commit:
Correct config reader

Commit:
Correct config reader 2

Commit:
Correct config reader 3

Commit:
Correct config reader 4

eaidova committed:
AC: fix IR loading

eaidova committed:
Merge pull request #750 from eaidova/ea/fix_ir_loading

AC: fix IR loading

vladimir-dudnik committed:
2019R4 models RC1

vladimir-dudnik committed:
Merge pull request #751 from vladimir-dudnik/r4-models-rc1

2019R4 models RC1

vladimir-dudnik committed:
Merge pull request #693 from vladimir-dudnik/fix-models-docs

remove end of lifed models from index and folders tree

vladimir-dudnik committed:
remove old models from test cases

vladimir-dudnik committed:
fix binary models subfolder

vladimir-dudnik committed:
Merge pull request #752 from vladimir-dudnik/fix-test-cases

remove old models from test cases

eaidova committed:
AC configs: adopt to new layout (#736)

* AC configs: adopt to new layout

* add fp32-int8 according to RC1 content

* resnet18-xnor

vladimir-dudnik committed:
Merge pull request #747 from Wovchena/demos-cnnnetreader-weights

demos: update cnnnetreader weights reading

ilyachur committed:
All demos are migrated to new API

Wovchena committed:
Test demos on new reid, update reid with output of 512

vshampor committed:
Add ResNet18 (v1, PyTorch)

vladimir-dudnik committed:
Merge pull request #758 from Wovchena/reid-models

Test demos on new reid, update reid with output of 512

Commit:
update review

eaidova committed:
AC: update config for brain-tumor-segmentation-0001 (#759)

Commit:
Renamed demo's directory

Commit:
Added models.lst

Commit:
renamed colorization.py

Commit:
Update colorization README.md

Commit:
Update colorization demo

Commit:
Update colorization demo 2

nestolen committed:
Update config_reader.py

Commit:
update demo 3

Commit:
update demo 4

eaidova committed:
AC: fix initialization order (#761)

Commit:
update demo 5

Commit:
update demo: fixed batch size = 1

Commit:
update demo: remove network info

Commit:
update demo: renamed exception

Commit:
Added option to show example of gesture for ASL demo

vshampor committed:
Update models/public/resnet-18-pytorch/model.yml

Co-Authored-By: Katya <ekaterina.aidova@intel.com>

Commit:
remove custom mean

Commit:
Update the upcoming release version

Commit:
requirements-pytorch.in: work around a torchvision bug

Commit:
Merge pull request #765 from IRDonch/version

Update the upcoming release version

Commit:
update demo: remove scale values

Wovchena committed:
instance_segmentation: fix

For Ubuntu16 the demo drew wrong masks sometimes.
I was deleting and returning back parts of the demo's code to find what causes it.
It appeared, that deleting the line `del net` fixes the issue.
Delete `del exec_net` and `del ie` just because they are not really needed.

Commit:
update demo: rewrite description

Commit:
update demo: renamed description 2

Commit:
update demo: update README.md

ilyachur committed:
Fixed comments

FengYen-Chang committed:
add ctc beam search decoder

Commit:
Merge pull request #766 from IRDonch/pillow-6

requirements-pytorch.in: work around a torchvision bug

Commit:
Fix typo: ouput

Commit:
Fix typo: embedings

Commit:
Merge pull request #769 from IRDonch/typos

Fix typos

Commit:
face-recognition-mobilefacenet-arcface: fix the documented output size

eaidova committed:
AC configs: replace daaset for person-detection-asl-0001 (#772)

aalborov committed:
Initial review

eaidova committed:
AC configs: set U8 input precision for semantic-segmentation-adas-0001 (#775)

Commit:
Merge pull request #767 from Wovchena/demos-instance_segmentation-fix

instance_segmentation: fix

Commit:
text-image-super-resolution-0001: fix the documented input shape

Commit:
Replace a deprecated MO option with its successor in MO configs

Commit:
Publish new versions of Intel models

These should fix the demo tests.

Some of the *-INT8 models are not available due to known issues with quantization.
They may be uploaded later.

Commit:
Merge pull request #778 from IRDonch/replace-deprecated-option

Replace a deprecated MO option with its successor in MO configs

eaidova committed:
AC: fix nifti reader (#779)

* AC: fix nifti reader

* readmes update

vladimir-dudnik committed:
Merge pull request #780 from IRDonch/new-models

Publish new versions of Intel models

FengYen-Chang committed:
use struct instead class and move to internal, rename paras and func

FengYen-Chang committed:
replace 'logic_error' by 'invalid_argument'

FengYen-Chang committed:
use std::min replace if/else

druzhkov-paul committed:
update AC config and docs for person-detection-asl-0001

ilyachur committed:
Update demos/human_pose_estimation_demo/src/human_pose_estimator.cpp

Co-Authored-By: Roman Donchenko <roman.donchenko@intel.com>

Commit:
Merge pull request #771 from IRDonch/arcface-size

face-recognition-mobilefacenet-arcface: fix the documented output size

Ilya-Krylov committed:
use output name from network info

Ilya-Krylov committed:
removed extra normalization as it is done in the network

Commit:
Merge pull request #762 from ilyachur/feature/ichuraev/fix_deprecation

Fixed deprecation warnings

Commit:
Merge pull request #783 from IRDonch/tisr-input-size

text-image-super-resolution-0001: fix the documented input shape

Commit:
Merge pull request #784 from Ilya-Krylov/fix/ik/image_retrieval_demo

Fix/ik/image retrieval demo

FengYen-Chang committed:
patch the string to vector<int> and check the prob_blank to avoid the logic error

FengYen-Chang committed:
change bandwith type from int to uint

FengYen-Chang committed:
add ctc description

Wovchena committed:
pedestrian_tracker: disable dyn batch

Wovchena committed:
crossroad_camera: replace person-reidentification-retail 0079 with 0031

person-reidentification-retail-0079 was removed.

Commit:
Fixed comments

Commit:
Merge pull request #786 from Wovchena/crossroad_camera-update-model-used

crossroad_camera: replace person-reidentification-retail 0079 with 0031

aalborov committed:
Fixes

aalborov committed:
fix

Commit:
Fixed multiple output network processing

aalborov committed:
Update product-detection-0001.md

Commit:
Update metrics for model 0200

aalborov committed:
Update text-spotting-0001-detector.md

aalborov committed:
Update efficientnet-b0-pytorch.md

aalborov committed:
Update efficientnet-b0.md

aalborov committed:
2 more models

FengYen-Chang committed:
remove comments and make last candidate as const

FengYen-Chang committed:
add more detail explanation for flag '-b'

FengYen-Chang committed:
change function softmax

eaidova committed:
AC: fix brain-tumor-segmentation-0002 (#787)

eaidova committed:
AC configs: fix missed metrics section in road-segmentation-adas-0001 config (#789)

Commit:
Merge pull request #785 from Wovchena/pedestrian_tracker-disable-dyn-batch

pedestrian_tracker: disable dyn batch

aalborov committed:
Fixes

aalborov committed:
Fixes

aalborov committed:
Fixes

aalborov committed:
Fixes

aalborov committed:
Update text-spotting-0001-detector.md

Commit:
Merge branch 'develop' of https://github.com/opencv/open_model_zoo into an/colorization-net

Commit:
update model: added accuracy

Commit:
Fixed comments

aalborov committed:
fix

aalborov committed:
Merge branch 'develop' of https://github.com/aalborov/open_model_zoo into develop

Commit:
Merge pull request #773 from aalborov/develop

Documentation Review

Commit:
update AC: update path format

Commit:
object_detection_demo_centernet: make compatible with NumPy < 1.15

NumPy 1.15 is when the axis parameter was made optional.

The NumPy version in the Ubuntu 18.04 repositories is 1.13, so it would
be good to be compatible with that.

Commit:
Merge pull request #763 from evgeny-izutov/feature/ei/asldemo_examples

ASL gesture recognition demo enhancement

Commit:
single-human-pose-estimation-0001: use correct base dirs in the config

Commit:
Merge pull request #794 from IRDonch/shpe-dl-dir

single-human-pose-estimation-0001: use correct base dirs in the config

Commit:
Merge pull request #793 from IRDonch/centernet-flip

object_detection_demo_centernet: make compatible with NumPy < 1.15

eaidova committed:
fix emotions-recognition-retail-0003 naming in description

Commit:
Merge pull request #796 from eaidova/ea/emotion_recognition_bug

fix emotions-recognition-retail-0003 naming in description

vladimir-dudnik committed:
Merge pull request #788 from DmitriySidnev/reid/0200/update_description

Update metrics for model 0200

vladimir-dudnik committed:
Merge pull request #782 from druzhkov-paul/update/dp/update_config_for_person_detection_asl_0001

Update for person-detection-asl-0001

Commit:
Publish new versions of Intel models

Because of improvements in the quantization tools, the following models are
now available in FP32-INT8:

- `instance-segmentation-security-0050`
- `instance-segmentation-security-0083`
- `person-detection-retail-0002`
- `text-image-super-resolution-0001`

Because of a remaining quantization issue, `person-attributes-recognition-crossroad-0230`
is no longer available in FP32-INT8.

Commit:
Merge pull request #797 from IRDonch/new-models

Publish new versions of Intel models

Commit:
Remove cpu_extension compatibility code

It's no longer needed with the current Inference Engine.

FengYen-Chang committed:
optimize

Commit:
update accuracy metrics and description

eizamaliev committed:
R-FCN-Resnet-101 from TensorFlow

Wovchena committed:
demos/monitors: beautify

Make plot lines reach edges of graphs.
Make graph titles darker to contrast with light background.w

Commit:
update demo: update materics value

nestolen committed:
Update models/public/colorization-v2-norebal/colorization-v2-norebal.md

Co-Authored-By: eizamaliev <48511626+eizamaliev@users.noreply.github.com>

nestolen committed:
Update models/public/colorization-v2/colorization-v2.md

Co-Authored-By: eizamaliev <48511626+eizamaliev@users.noreply.github.com>

Commit:
Bump the version for the next release

Commit:
Merge pull request #801 from IRDonch/ver-2020.2

Bump the version for the next release

Commit:
Merge pull request #768 from FengYen-Chang/CTCBS

Add ctc beam search decoder for text_detection demo

Commit:
Merge pull request #802 from IRDonch/bye-cpu-ext

Remove cpu_extension compatibility code

Commit:
Merge pull request #760 from opencv/vs/resnet18_config

Add ResNet18 accuracy checker config

Commit:
Revert "requirements-pytorch.in: work around a torchvision bug"

The workaround is no longer necessary with the release of torchvision 0.5.0.

This reverts commit db3f5af7cc7d070754b0bc6a68440bb365be07fd.

Commit:
Bump all dependency versions

ilya-lavrenov committed:
Fixed compilation on CentOS

eaidova committed:
AC: extract formatted result and metric names from presenter (#764)

* AC: extract formatted result and metric names from presenter

* make flatten output

* bug fix

eaidova committed:
AC: integrate Fashion MNIST annotation converter (#781)

* AC: integrate Fashion MNIST annotation converter

* fix typo

Commit:
add ac configs

Commit:
fix paddings

Commit:
Merge pull request #651 from IRDonch/faster-rcnn-simplification

Simplify post-processing in object_detection_demo_faster_rcnn

Commit:
fixes

Commit:
Merge pull request #804 from ilya-lavrenov/il/fix-centos

Fixed compilation on CentOS

Commit:
AC: don't use the deprecated np.matrix class

eaidova committed:
AC: update requirements for numpy

Commit:
Merge pull request #806 from eaidova/ea/numpy_req

AC: update requirements for numpy

Wovchena committed:
segmentation: update

Rewrite the demo to use OpnCV.
Delete the stuff that became unused after rewriting.

Wovchena committed:
segmentation: fix for Windows

Wovchena committed:
segmentation: fix README.md and the test

Apply suggestions from code review

Co-Authored-By: Roman Donchenko <roman.donchenko@intel.com>

Wovchena committed:
segmentation: add using namespace InferenceEngine; update README.md

eaidova committed:
Merge pull request #808 from IRDonch/np-matrix

AC: don't use the deprecated np.matrix class

eaidova committed:
Merge pull request #805 from jkamelin/jk/ac_configs

Add ac configs for googlenet-v4, resnet-50

Commit:
Merge pull request #803 from IRDonch/dependency-updates

Bump all dependency versions

Commit:
AC: fix a divide-by-zero warning in AverageMeter

We're dividing by `increment`, so that's what we should be checking. This was
probably a copy-paste error from `evaluate`.

eaidova committed:
Merge pull request #809 from IRDonch/divide-by-zero

AC: fix a divide-by-zero warning in AverageMeter

Wovchena committed:
Update demos/segmentation_demo/main.cpp

Co-Authored-By: Roman Donchenko <roman.donchenko@intel.com>

vladimir-dudnik committed:
add voc2012 without background to dataset definitions

eaidova committed:
Merge pull request #811 from opencv/dataset_definitions

add voc2012 without background to dataset definitions

Wovchena committed:
demos/monitors: fix

eaidova committed:
AC: support CIFAR100 dataset

eaidova committed:
Merge pull request #812 from eaidova/ea/cifar

AC: support CIFAR100 dataset

eizamaliev committed:
Descriptions for public modes are auto updated

Commit:
add models

Commit:
Merge pull request #800 from Wovchena/demos-monitors-beautify

demos/monitors: beautify

eaidova committed:
AC: extend NLP support (#757)

eaidova committed:
AC: sharing bitstream from previous run (#777)

eizamaliev committed:
Model added to index

eaidova committed:
AC: remove durty hack for channels alignment in launcher (#817)

* AC: remove durty hack for channels alignment in launcher

* readme

Wovchena committed:
segmentation: update comments, add -delay, report latency

Commit:
Merge pull request #815 from eizamaliev/r_fcn

R-FCN-Resnet-101 from TensorFlow

eaidova committed:
AC: retrun back to numpy 1.17.5 (#819)

Commit:
mask_rcnn_demo: use the documented output of the models

The documented output is `reshape_do_2d`, not `detection_output`. The current
code has worked so far, because a layer named `detection_output` also exists
in the network; however, it's an internal layer and we shouldn't be relying
on it.

Note that currently, `detection_output` is also marked by MO as an output
(unless you use IR v7). This might, however, be a bug. An internal report
already exists for it.

Commit:
demos: Remove unnecessary line continuations

They annoy me.

eaidova committed:
AC: separate dependencies installation (#816)

eaidova committed:
AC: support SRGAN (#821)

Commit:
add googlenet v1 config

Commit:
add links

Commit:
fix

Commit:
fix paths

Commit:
fix paths

eaidova committed:
Fix pillow resize (#824)

* AC: support SRGAN (#821)

* fix resize

eaidova committed:
AC: revise result extraction (#822)

Commit:
fixes

Commit:
demos: fix various Cppcheck style warnings

Commit:
human_pose_estimation_demo: replace asctime with strftime

asctime is marked obsolete in POSIX. Also, this allows us to use a date format
that is more appropriate for a timestamp (ordering the fields from slowest-
to fastest-changing).

Commit:
demos: drop unused functions

Commit:
security_barrier_camera_demo: make InferRequestsContainer uncopyable

Cppcheck complains about a violation of the rule of three. I could add
a real copy constructor; however, the assignment operator isn't really
needed either, so instead I deleted them both and implemented an `assign`
method, which is all the class really needs.

This makes the non-default constructor unnecessary, so I removed it too.

Commit:
pedestrian_tracker_demo: remove unused variables

eaidova committed:
AC: added support numpy files for action recognition (#825)

Commit:
Merge pull request #356 from allnes/an/colorization-net

Add colorization network

Commit:
Merge pull request #807 from Wovchena/segmentation-update

segmentation: update

Commit:
Merge pull request #820 from IRDonch/reshape-do-2d

mask_rcnn_demo: use the documented output of the models

Commit:
Merge pull request #828 from IRDonch/rm-line-continuations

demos: Remove unnecessary line continuations

Commit:
Merge pull request #813 from eizamaliev/doc_update

[DOC] Descriptions for public modes are auto updated

Commit:
Merge pull request #829 from IRDonch/cppcheck

demos: Fix various Cppcheck warnings

Commit:
downloader: remove the --config option

It's been deprecated for two releases now; time to get rid of it.

Wovchena committed:
security_barrier_camera: threadPull->threadPool, update README.md

Commit:
AC: Move the test requirement list to a file

The "setup.py test" command is deprecated (see
https://github.com/pypa/setuptools/pull/1878). I'd like to transition
CI to running pytest directly, but for that, I need an easy way to install
test dependencies. This provides such a way.

mzuevx committed:
IMDB annotation for XLNet accuracy test fixes. (#831)

* fix path for Sentence Piece model loading (PosixPath->str)
* fix method name (PieceTold -> PieceToId)
* fix convert_single_example() to be able  to work without sample.text_b

Commit:
Merge pull request #832 from Wovchena/security_barrier_camera-wording

security_barrier_camera: threadPull->threadPool, update README.md

Commit:
Merge pull request #830 from IRDonch/rm-big-config

downloader: remove the --config option

Commit:
Add a file with pinned versions of Accuracy Checker's test dependencies

pytest and pytest-mock are already pinned in the .in file, but they have
a boatload of dependencies, so it would be safer to pin them all.

Note that `requirements-ac-test.txt` is built from _both_ `requirements.in`
and `requirements-test.in`. This is so that if main and test dependencies
ever overlap, `requirements-ac-test.txt` will respect any restrictions
set in `requirements.in`. The test dependencies have to be installed into
the same virtual environment as the main dependencies, so we can't have any
conflicts between them.

It is annoying to have duplicated dependencies in two files, but since
they're generated, I don't think it's a big problem. One alternative would
be to just integrate the test dependencies into `requirements-ac.txt`,
but we might add some build jobs that use AC without also testing it,
and I'd rather not install the test dependencies in that case.

I also update `requirements-ac.txt` to keep the versions in both files
synchronized.

Commit:
Merge pull request #834 from IRDonch/ac-test-deps

AC: Create requirements files for test dependencies

eaidova committed:
AC: fix CIFAR100 conversion (#835)

Commit:
improved docs

vladimir-dudnik committed:
add unet-camvid-onnx-0001 description

Commit:
updates

Commit:
demos/README.md: fix broken link to the CenterNet demo

jkamelin committed:
set tf version

jkamelin committed:
set tf version

Commit:
Merge pull request #841 from IRDonch/centernet-link

demos/README.md: fix broken link to the CenterNet demo

Commit:
Merge pull request #823 from jkamelin/jk/2020.1_public_models

Add googlenet-v1/v4, resnet-50 TF models

Commit:
AC: Make tests compatible with latest pytest

The string representation of ExceptionInfo has changed. Use the exception
itself instead.

Commit:
Update Accuracy Checker test dependencies

Loosen the requirements in requirements-test.in, since CI no longer uses it
directly.

Commit:
AC: Set the pytest JUnit report format to 2

It's going to be the default in pytest 6, and pytest 5 warns about this.
We should be OK with version 2, so let's update now.

vladimir-dudnik committed:
add GFLOPs and MParams info

vladimir-dudnik committed:
add GFlops/MParams to model description

Commit:
Updated ASL Recognition model to the new one

eaidova committed:
AC: handle paths in tests OS agnostic

Commit:
Merge pull request #846 from eaidova/ea/ac_tests_win

AC: handle paths in tests OS agnostic

Commit:
Merge pull request #844 from IRDonch/update-pytest

AC: Switch to pytest 5

Commit:
downloader: remove unused function parameters

Commit:
downloader: hash files while they're downloading

This avoid rereading the entire file from disk once it's written.

Commit:
Replace unintentional Cyrillic C characters with Latin ones

Commit:
Replace unnecessary non-ASCII characters with ASCII ones

Or, in the case of the quotes around "The PASCAL Visual Object Classes (VOC)
Challenge", just remove them entirely, because they aren't needed there.

Commit:
AC: Replace non-ASCII characters with escapes

This has a few small benefits:

1. There's no possibility of the editor misdetecting the file encoding
   and displaying the source incorrectly.

2. For esoteric characters, names are easier to understand than
   the characters themselves.

3. It's easier to distinguish between similar-looking characters this way.

Commit:
Add a script to update ci/requirements-*.txt

This automates the process I've been doing manually so far.
This will make it easier, and will enable everyone else to do it too.

eaidova committed:
AC: introduce config parameter for isolated location segmentation masks from images (#840)

eizamaliev committed:
Added "cells" parameter to yolo-v2 adapter (#849)

eaidova committed:
AC: fix input transformation for FPGA (#850)

Commit:
Merge pull request #847 from IRDonch/hash-while-downloading

downloader: hash files while they're downloading

eaidova committed:
[Public models]: fix typo in model.yml

Commit:
Merge pull request #851 from eaidova/fix_reverse_input_channel

[Public models]: fix typo in model.yml

Commit:
demos: fix the languages specified for code blocks in the docs

Mostly, just remove "sh" from blocks with help text, since they are not
in any formal language. But there are also a few cases where a block
was labeled with the wrong language.

eaidova committed:
AC: fix annotation meta

Commit:
Merge pull request #839 from vladimir-dudnik/vd/unet-camvid-onnx-0001

add unet-camvid-onnx-0001 description

Commit:
Merge pull request #848 from IRDonch/asciify

Remove non-ASCII characters

Commit:
Merge branch 'develop' into sbcd-hddl-hint

Commit:
security_barrier_camera_demo/README.md: reformat the VAD hints

Also, improve the wording a bit.

Commit:
Merge pull request #854 from IRDonch/sbcd-hddl-hint

security_barrier_camera_demo: add optimization hints for Intel VAD

Commit:
Merge pull request #853 from IRDonch/code-block-langs

demos: fix the languages specified for code blocks in the docs

Commit:
Merge pull request #855 from IRDonch/update-reqs

Add a script to update ci/requirements-*.txt

eaidova committed:
fix dummy launcher (#857)

Commit:
Merge pull request #845 from evgeny-izutov/feature/ei/update_aslnet_model

Updated ASL Recognition model to the new one

FenixFly committed:
Merge branch 'develop' into Smart-Library
